src/main/java/com/amazon/aocagent/services/ECSService.java

```diff
@@ -0,0 +1,211 @@
+    return ecsClient.createCluster(request);
+  }
+
+  /**
+   * create and register ecs task definition for both EC2 and fargate running mode.
+   * @param taskDef ecs task definition
+   * @return {@link RegisterTaskDefinitionResult} for registering ecs container defs
+   */
+  public RegisterTaskDefinitionResult registerTaskDefinition(String taskDef) throws BaseException {
+    RegisterTaskDefinitionRequest request;
+
+    try {
+      request = jsonMapper.readValue(taskDef, RegisterTaskDefinitionRequest.class);
+    } catch (Exception e) {
+      throw new BaseException(ExceptionCode.ECS_TASK_EXECUTION_FAIL, e.getMessage());
+    }
+    return ecsClient.registerTaskDefinition(request);
+  }
+
+  /**
+   * create and run ECS task definitions based on the launch type.
+   * @param runTaskRequest ecs run task request
+   */
+  public void runTaskDefinition(RunTaskRequest runTaskRequest) throws BaseException {
+
+    final RunTaskResult runTaskResult = ecsClient.runTask(runTaskRequest);
+    runTaskResult
+        .getTasks()
+        .forEach(task -> log.info("Successfully running task [{}]", task.getTaskArn()));
+
+    if (!runTaskResult.getFailures().isEmpty()) {
+      throw new BaseException(
+          ExceptionCode.ECS_TASK_EXECUTION_FAIL, runTaskResult.getFailures().toString());
+    }
+  }
+
+  /**
+   * clean ECS tasks resources.
+   * @throws InterruptedException fail to clean exception
+   */
+  public void cleanTasks() throws Exception {
+    List<String> clusters = this.listClusters();
+    for (String clusterName : clusters) {
+      // clean up tasks
+      for (String taskArn :
+          ecsClient
+              .listTasks(
+                  new ListTasksRequest()
+                      .withCluster(clusterName)
+                      .withDesiredStatus(DesiredStatus.RUNNING))
+              .getTaskArns()) {
+        ecsClient.stopTask(new StopTaskRequest().withTask(taskArn).withCluster(clusterName));
+      }
+
+      // make sure there is no RUNNING status task remaining
+      RetryHelper.retry(
+          () -> {
+            if (ecsClient
+                .listTasks(
+                    new ListTasksRequest()
+                        .withCluster(clusterName)
+                        .withDesiredStatus(DesiredStatus.RUNNING))
+                .getTaskArns()
+                .isEmpty()) {
+              return;
+            }
+            TimeUnit.SECONDS.sleep(Integer.parseInt(GenericConstants.SLEEP_IN_MILLISECONDS.getVal()));
+          });
+
+      log.info("finish metric validation");
+    }
+  }
+
+  /**
+   * clean ECS cluster.
+   */
+  public void cleanCluster() {
+    for (String name : listClusters()) {
+      DeleteClusterRequest request = new DeleteClusterRequest().withCluster(name);
+      try {
+        ecsClient.deleteCluster(request);
+      } catch (Exception e) {
+        log.error("{} can't be deleted", name, e);
+      }
+    }
+  }
+
+  private List<String> listClusters() {
```

Comment:
does it mean we will clean all the clusters even it's running?

Reply:
Yes, I forgot to only filter out the clusters with our prefix.  I was planed to do it before coding. lol